### DISCUSSION TITLE
BGP Summary and Neighbor

### DISCUSSION THREAD

**Discussion by brickbr** [thatmattlove/hyperglass#95] [Ideas]
May I suggest adding the ability to run the command (Cisco) "Show BGP ipv4/ipv6 unicast summary" for a quick list of all BGP sessions on a router. I would suggest since this command does not take input (other than VRF in a VRF situation) to grey out the input field.

Finally may I suggest a feature to run Neighbor specific commands, like (cisco) "show bgp neighbor x.x.x.x detail".

I find that once i have details about the current state of my tables using the route commands, I am logging into the router to gather more information for a peer, which could be reduced with the above commands, keeping me in the looking glass tool longer.

Thanks,
Brock Richards

**Comment by thatmattlove** [comment]
Hey there,

I think this is a great idea, however there are a few considerations. Namely, we need to determine the appropriate commands for the desired output on each [natively supported platform](https://hyperglass.io/docs/introduction). I have easy access to EOS, BIRD, IOS-XR, IOS/IOS-XE, FRR, Junos, Mikrotik, TNSR, and VyOS and could determine this for those (though, it would probably take a little while if I'm the only one testing those platforms). I do not, however, have access to Huawei. Those commands were provided by  @glaubway in #17.

Is this something you'd be able to help with? To add support for other commands, we would need to determine the IPv4 & IPv6 commands for each device, or if dual stack, those commands. Additionally, and caveats across platforms, such as "juniper only accepts the query target in x format, whereas everything else excepts in y format", therefore requiring some kind of parsing/manipulation before sending it on to the device.

If this is something you'd be willing to help with, go ahead and open an issue for it, and you can post any commands you're able to find there, and I'll do the same.

**Reply by kastakhov** [reply]
Hi, 
For Huawei you can use the following commands.

List of all ipv4 bgp peers:
`display bgp peer`
List of all ipv6 bgp peers:
`display bgp ipv6 peer`

List detail information about ipv4 bgp peer:
`display bgp peer <ip> verbose`
List detail information about ipv6 bgp peer:
`display bgp ipv6 peer <ip> verbose`

Where `<ip>` is the ip of your bgp neighbor.

**Comment by warriorsoul15** [comment]
nxos :  show ip bgp summary

also would love to see advertised and received routes from a peer.

show bgp ipv4 unicast neighbor X.X.X.X advertised-routes
show bgp ipv4 unicast neighbor X.X.X.X received-routes
